src/gwt/src/org/rstudio/core/client/command/AppCommand.java

```diff
@@ -186,6 +184,9 @@ public void setEnabled(boolean enabled)
          enabled_ = enabled;
          handlers_.fireEvent(new EnabledChangedEvent(this));
       }
+    
+      if (Desktop.isDesktop())
```

Comment:
We might want to make this macOS-only (?)

Reply:
Why Mac-only?

Reply:
Mainly just because this is the behaviour we've had on other platforms, and it's worked as expected.

(And yet, I'm somewhat surprised that it works as expected without these calls...)

Reply:
The reason it worked as expected earlier on macOS is that we had a hook that ran prior to the menu showing which did a just-in-time sync of the command state. 

https://github.com/rstudio/rstudio/blob/bf986733d06237283adb8c67223123a8b3b058cf/src/cpp/desktop-mac/MainFrameMenu.mm#L218-L244

Reply:
In 1.2 we changed this; because callbacks into JS are asynchronous but the menu is shown synchronously, we can't delay showing it until we collect the command state from the client (pull model). Instead we send the command state from the client to the desktop frame as it changes (push model).